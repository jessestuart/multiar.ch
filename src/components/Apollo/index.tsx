import ApolloClient from 'apollo-boost'
import fetch from 'isomorphic-fetch'

export const client = new ApolloClient({
  fetch,
  uri: 'https://dh.jesses.dev/api/',
})

export default client
