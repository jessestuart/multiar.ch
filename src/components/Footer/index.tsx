import React from 'react'
import Theme from 'styles/Theme'
import { DateTime } from 'luxon'
import { Heart } from 'react-feather'

const { colors } = Theme

const Footer = () => {
  return (
    <footer
      className="bb bottom-0 bw2 f5-ns f6 flex flex-column fw4 lh-title pa4 w-100"
      style={{
        borderColor: colors.primary,
        borderTop: `1px #e4e4e4 solid`,
      }}
    >
      <div className="flex justify-center items-center">
        © 2018–{DateTime.local().year} Jesse Stuart
      </div>
      <div className="dib center tc flex-ns justify-center items-center">
        Pixels, bits and bytes hand-crafted with
        <Heart
          className="hot-pink mh1 hvr-grow"
          fill="#ff3a5c"
          height="15"
          width="15"
        />
        in NYC.
      </div>
    </footer>
  )
}

export default Footer
