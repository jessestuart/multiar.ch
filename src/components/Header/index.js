/* @flow */
import React from 'react'
import Link from 'gatsby-link'

type Props = {
  siteTitle: string,
}

const Header = ({ siteTitle }: Props) => {
  return (
    <div
      className="pv3"
      style={{
        background: '#e5f4ee',
      }}
    >
      <div
        className="flex justify-center items-center"
        style={{
          margin: '0 auto',
          maxWidth: 960,
        }}
      >
        <h1 className="ma0 mt2 f1">
          <Link
            to="/"
            className="fw3 no-underline"
            style={{
              textDecoraction: 'none',
              color: '#68465b',
              letterSpacing: '-0.15rem',
            }}
          >
            {siteTitle}
          </Link>
        </h1>
      </div>
    </div>
  )
}

export default Header
