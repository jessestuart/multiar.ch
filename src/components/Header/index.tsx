import Link from 'gatsby-link'
import React, { ReactNode } from 'react'
import { Flex, Text } from 'rebass'

interface Props {
  children: ReactNode
}

const Header = ({ children }: Props) => (
  <Flex bg="secondary" className="justify-center items-center center w-100 pa2">
    <Link to="/" className="fw3 no-underline tracked-tight">
      <Text color="primary" fontFamily="mono" className="ma0 mt2 f1">
        {children}
      </Text>
    </Link>
  </Flex>
)

export default Header
