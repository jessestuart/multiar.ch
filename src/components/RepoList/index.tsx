import { graphql, useStaticQuery } from 'gatsby'
import { DockerHubRepo } from 'gatsby-source-docker-hub'
import _ from 'lodash'
import fp from 'lodash/fp'
import { DateTime } from 'luxon'
import React from 'react'
import { Box, Text } from 'rebass'

import Theme from 'styles/Theme'

const RepoList = () => {
  const query = useStaticQuery(graphql`
    {
      allDockerHubRepo {
        edges {
          node {
            description
            id
            lastUpdated
            name
            pullCount
            starCount
          }
        }
      }
    }
  `)

  const repos: DockerHubRepo[] = _.flow(
    fp.get('allDockerHubRepo.edges'),
    fp.map('node'),
    // fp.filter(repo => DateTime.fromISO(repo.lastUpdated).diffNow()),
  )(query)
  if (_.isEmpty(repos)) {
    return null
  }

  return (
    <Box className="justify-center center flex pt4 f4">
      <ol className="mw7-ns w-100">
        <li className="flex fw7 lh-copy">
          <Text className="flex-nowrap w-100" fontFamily="mono">
            Image
          </Text>
          <Text className="flex justify-end w-100" fontFamily="mono">
            Pulls
          </Text>
        </li>
        {repos.map((repo: DockerHubRepo, index: number) => {
          const lastUpdatedRelative = DateTime.fromISO(
            _.toString(repo.lastUpdated),
          ).toRelative()

          return (
            <li key={index} className="flex lh-copy pt2 pb3 bb b--near-white">
              <div className="w-100">
                <a
                  className="flex-nowrap primary lh-title"
                  style={{ color: Theme.colors.primary }}
                  href={`https://cloud.docker.com/u/jessestuart/repository/docker/jessestuart/${repo.name}`}
                >
                  <Text fontFamily="mono" color="primary" className="fw5">
                    {repo.name}
                  </Text>
                </a>
                <Text>{repo.description}</Text>
                <Text
                  className="flex justify-start no-underline flex-column f6"
                  fontFamily="sans-serif"
                  style={{ color: Theme.colors.textMuted }}
                >
                  Updated {lastUpdatedRelative}
                </Text>
              </div>
              <Text
                className="pl4"
                style={{ display: 'inline-block' }}
                fontFamily="mono"
              >
                {repo.pullCount.toLocaleString()}
              </Text>
            </li>
          )
        })}
      </ol>
    </Box>
  )
}

export default RepoList
