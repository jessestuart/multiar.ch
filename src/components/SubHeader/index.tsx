import React from 'react'
import { Text } from 'rebass'

const SubHeader = () => {
  return (
    <div className="pv4 bb bt b--black-10 flex justify-center">
      <Text className="flex flex-column lh-title mw7">
        <Text
          fontFamily="serif"
          className="i justify-center center pl3"
          style={{
            borderLeft: '2px rgba(104, 70, 91, 0.7) solid',
          }}
        >
          Cross-build so hard, muhfuckas wanna fine me;
          <br />
          (that manifest cray)
        </Text>
        <div className="flex justify-end">
          <Text fontFamily="serif" pl={2}>
            â€” Jay-Z, Probably
          </Text>
        </div>
      </Text>
    </div>
  )
}

export default SubHeader
