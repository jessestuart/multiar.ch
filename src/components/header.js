/* @flow */
import React from 'react'
import Link from 'gatsby-link'

type Props = {
  siteTitle: string,
}

const Header = ({ siteTitle }: Props) => {
  console.log({ siteTitle })
  return (
    <div
      className="pv3"
      style={{
        background: '#e5f4ee',
      }}
    >
      <div
        className="flex justify-center items-center"
        style={{
          margin: '0 auto',
          maxWidth: 960,
        }}
      >
        <h1 className="ma0 mt2">
          <Link
            to="/"
            className="black-70 fw3 no-underline"
            style={{ textDecoraction: 'none' }}
          >
            {siteTitle}
          </Link>
        </h1>
      </div>
    </div>
  )
}

export default Header
