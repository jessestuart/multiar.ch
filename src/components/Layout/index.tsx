import 'js-tachyons'
import 'typeface-lato'
import 'typeface-space-mono'

import React, { ReactNode } from 'react'
import Helmet from 'react-helmet'
import { Box } from 'rebass'

import Header from 'components/Header'
import { useSiteMetadata } from 'utils/hooks'

interface Props {
  children: ReactNode
}

const Layout = ({ children }: Props) => {
  const siteMetadata = useSiteMetadata()
  const { title } = siteMetadata
  return (
    <>
      <Helmet title={title} />
      <Header>{title}</Header>
      <Box className="f3 w-100-ns ma4 ma0-ns lh-copy">{children}</Box>
    </>
  )
}

export default Layout
