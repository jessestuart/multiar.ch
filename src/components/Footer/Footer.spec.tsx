import React from 'react'
import renderer from 'react-test-renderer'

import { Copyright, Footer, MadeWithLove } from 'components/Footer'

describe('Footer components.', () => {
  test('Copyright component.', () => {
    const component = renderer.create(<Copyright />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('MadeWithLove component.', () => {
    const component = renderer.create(<MadeWithLove />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Footer component.', () => {
    const component = renderer.create(<Footer />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
