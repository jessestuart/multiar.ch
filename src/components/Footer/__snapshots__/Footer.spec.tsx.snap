// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer components. Copyright component. 1`] = `
<div
  className="justify-center items-center css-4cffwv"
>
  © 2018–
  2020
   
  <a
    className="pl1 b"
    href="https://jesses.io/"
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.75)",
      }
    }
  >
    Jesse Stuart
  </a>
</div>
`;

exports[`Footer components. Footer component. 1`] = `
<footer
  className="b-primary bb bottom-0 bw2 f5-ns f6 flex flex-column fw4 pa4 w-100"
  style={
    Object {
      "borderTop": "1px rgba( 0, 0, 0, 0.1) solid",
    }
  }
>
  <div
    className="justify-center items-center css-4cffwv"
  >
    © 2018–
    2020
     
    <a
      className="pl1 b"
      href="https://jesses.io/"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.75)",
        }
      }
    >
      Jesse Stuart
    </a>
  </div>
  <div
    className="dib center tc flex-ns justify-center items-center css-vurnku"
  >
    Pixels, bits and bytes hand-crafted with
    <svg
      className="hot-pink mh1"
      fill="#ff3a5c"
      height={16}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
      />
    </svg>
    in NYC.
  </div>
</footer>
`;

exports[`Footer components. MadeWithLove component. 1`] = `
<div
  className="dib center tc flex-ns justify-center items-center css-vurnku"
>
  Pixels, bits and bytes hand-crafted with
  <svg
    className="hot-pink mh1"
    fill="#ff3a5c"
    height={16}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    viewBox="0 0 24 24"
    width={16}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
    />
  </svg>
  in NYC.
</div>
`;
